{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Script written to compare VAF's between two VCF files\n",
    "#by: reykajayasinghe@gmail.com\n",
    "#Edited on: 04/09/2019\n",
    "\n",
    "#V1=\"/Volumes/Angie's Bac/Met Seq Data/Index35_OtB3992_Angela.Hirbe_900-1985a_human_hEx_AV5_30x_07222016.deduplicated.realigned.recalibrated.snpEff.snpSift.vcf\"\n",
    "#V2=\"/Volumes/Angie's Bac/Met Seq Data/Index36_OtB3993_Angela.Hirbe_900-1985a_human_hEx_AV5_30x_FFPE_07222016.deduplicated.realigned.recalibrated.snpEff.snpSift.vcf\"\n",
    "#V3=\"/Volumes/Angie's Bac/Met Seq Data/Index37_OtB3994_Angela.Hirbe_900-1985a_human_hEx_AV5_30x_FFPE_07222016.deduplicated.realigned.recalibrated.snpEff.snpSift.vcf\"\n",
    "#V4=\"/Volumes/Angie's Bac/Met Seq Data/Index38_OtB3995_Angela.Hirbe_900-1985a_human_hEx_AV5_30x_FFPE_07222016.deduplicated.realigned.recalibrated.snpEff.snpSift.vcf\"\n",
    "\n",
    "V1=\"/Volumes/Angie's Bac/Met Seq Data/Index39_OtB3996_Angela.Hirbe_900-1985a_human_hEx_AV5_30x_07222016.deduplicated.realigned.recalibrated.snpEff.snpSift.vcf\"\n",
    "V2=\"/Volumes/Angie's Bac/Met Seq Data/Index40_OtB3997_Angela.Hirbe_900-1985a_human_hEx_AV5_30x_FFPE_07222016.deduplicated.realigned.recalibrated.snpEff.snpSift.vcf\"\n",
    "V3=\"/Volumes/Angie's Bac/Met Seq Data/Index42_OtB3998_Angela.Hirbe_900-1985a_human_hEx_AV5_30x_FFPE_07222016.deduplicated.realigned.recalibrated.snpEff.snpSift.vcf\"\n",
    "V4=\"/Volumes/Angie's Bac/Met Seq Data/Index43_OtB3999_Angela.Hirbe_900-1985a_human_hEx_AV5_30x_FFPE_07222016.deduplicated.realigned.recalibrated.snpEff.snpSift.vcf\"\n",
    "\n",
    "#Index35_OtB3992_Angela.Hirbe_900-1985a_human_hEx_AV5_30x_07222016.deduplicated.realigned.recalibrated.snpEff.snpSift.vcf\n",
    "#Index36_OtB3993_Angela.Hirbe_900-1985a_human_hEx_AV5_30x_FFPE_07222016.deduplicated.realigned.recalibrated.snpEff.snpSift.vcf\n",
    "#Index37_OtB3994_Angela.Hirbe_900-1985a_human_hEx_AV5_30x_FFPE_07222016.deduplicated.realigned.recalibrated.snpEff.snpSift.vcf\n",
    "#Index38_OtB3995_Angela.Hirbe_900-1985a_human_hEx_AV5_30x_FFPE_07222016.deduplicated.realigned.recalibrated.snpEff.snpSift.vcf\n",
    "#Index39_OtB3996_Angela.Hirbe_900-1985a_human_hEx_AV5_30x_07222016.deduplicated.realigned.recalibrated.snpEff.snpSift.vcf\n",
    "#Index40_OtB3997_Angela.Hirbe_900-1985a_human_hEx_AV5_30x_FFPE_07222016.deduplicated.realigned.recalibrated.snpEff.snpSift.vcf\n",
    "#Index42_OtB3998_Angela.Hirbe_900-1985a_human_hEx_AV5_30x_FFPE_07222016.deduplicated.realigned.recalibrated.snpEff.snpSift.vcf\n",
    "#Index43_OtB3999_Angela.Hirbe_900-1985a_human_hEx_AV5_30x_FFPE_07222016.deduplicated.realigned.recalibrated.snpEff.snpSift.vcf\n",
    "\n",
    "#Dependencies\n",
    "import sys\n",
    "import os\n",
    "import subprocess\n",
    "from collections import defaultdict\n",
    "\n",
    "mutations_normal=defaultdict(float)\n",
    "mutations_t1=defaultdict(float)\n",
    "mutations_t2=defaultdict(float)\n",
    "mutations_t3=defaultdict(float)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Read through VCF File\n",
    "file=open(V4,\"r\")\n",
    "skippedindel=0\n",
    "skippedlowreadcount=0\n",
    "skippedlowvaf=0\n",
    "totalsites=0\n",
    "import sys\n",
    "\n",
    "for line in file:\n",
    "    line.strip()\n",
    "    skipvalue=0\n",
    "    if line.startswith('#'):\n",
    "        continue\n",
    "      # PROCESS LINE HERE\n",
    "    else:\n",
    "        chrom,position,ID,reference,variant,quality,ftype,info,formattype,sample=line.split(\"\\t\")\n",
    "        #CHECK 1: is it an indel, if so filter out for now\n",
    "        refsize=len(reference)\n",
    "        varsize=len(variant)\n",
    "        if refsize == varsize:\n",
    "            try:\n",
    "                (GT,AD,DP,GQ,PL)=sample.split(\":\")\n",
    "                #print(\"GOOD:\",line)\n",
    "                (refcount,varcount)=AD.split(\",\")\n",
    "                vaf=(float(varcount)/(float(refcount)+float(varcount)))*100\n",
    "                #print(sample,float(vaf),refcount,varcount)\n",
    "                if (float(vaf) < 5.0):\n",
    "                    #skippedlowvaf+=1\n",
    "                    #print(\"SKIP\",str(vaf))\n",
    "                    skipvalue=1\n",
    "                    continue\n",
    "                else:\n",
    "                    totalsites+=1\n",
    "                    final=chrom+\"\\t\"+position+\"\\t\"+refcount+\"\\t\"+varcount+\"\\t\"+str(vaf)\n",
    "                    positionfull=chrom+\"_\"+position+\"_\"+reference+\"_\"+variant\n",
    "                    #Store this position and associated VAF in dictionary to compare to next sample\n",
    "                    mutations_t3[positionfull]=vaf\n",
    "                    #print(final)\n",
    "            except ValueError:\n",
    "                #When there are two alt alleles - these are mostly indels\n",
    "                #BAD: chr2\t233345249\trs1190432\tC\tA,T\n",
    "                #In this case there are more values for GT,AD,DP,GQ,PL\n",
    "                #1/2:1,13,5:19:99:515,135,137,347,0,368\n",
    "                #The second element refers to the first alt allele and the second refers to the second alt allele count\n",
    "                # 13 - A variant, 5 - T variant, 19 read depth\n",
    "                #print(\"BAD:\",line)\n",
    "                skippedindel+=1\n",
    "                continue\n",
    "        else:\n",
    "            skippedindel+=1\n",
    "            \n",
    "print(skippedindel,skipvalue)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "normal=list(mutations_normal)\n",
    "t1=list(mutations_t1)\n",
    "t2=list(mutations_t2)\n",
    "t3=list(mutations_t3)\n",
    "\n",
    "unique_mutations=set(normal+t1+t2+t3)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "sys.stdout = open(\"/Users/rjayasin/Desktop/Mutation_S2_vafs.txt\",\"wt\")\n",
    "\n",
    "for mutation in unique_mutations:\n",
    "    normal = mutations_normal.get(mutation, \"0.0\")\n",
    "    t1 = mutations_t1.get(mutation, \"0.0\")\n",
    "    t2 = mutations_t2.get(mutation, \"0.0\")\n",
    "    t3 = mutations_t3.get(mutation, \"0.0\")\n",
    "    print(mutation,normal,t1,t2,t3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
